DOC-3209: TinyMCE 8.1.0 Release Documentation and Community Changelog. (#3824)
tinymce/tinymce-docs: modules/ROOT/pages/8.1.0-release-notes.adoc

[[overview]]
== Overview

{productname} {release-version} was released for {enterpriseversion} and {cloudname} on Wednesday, September 17^th^, 2025. These release notes provide an overview of the changes for {productname} {release-version}, including:

> * xref:new-premium-plugins[New Premium plugins]

Author: ```suggestion
* xref:new-premium-plugin[New Premium plugin]
```

**Spell Checker** Premium plugin includes the following fix.

==== Spellcheck could create an invalid selection if the first element in the document was `+noneditable+`.
// #TINY-12514

Previously when the first element of a document was set as a `contenteditable="false"` (CEF) element, selecting all content with {productname} spellchecker could result in an invalid selection state. This issue prevented the deletion of the entire content selection, creating an edge case that disrupted normal editing workflows.

To resolve this, {productname} now inserts a temporary hidden element before the initial CEF element in such scenarios. This adjustment ensures that full selections are valid and can be deleted as expected.

For information on the **Spell Checker** plugin, see: xref:introduction-to-tiny-spellchecker.adoc[Spell Checker].

=== Image Optimizer (Powered by Uploadcare)

The {productname} {release-version} release includes an accompanying release of the **Image Optimizer (Powered by Uploadcare)** premium plugin.

**Image Optimizer (Powered by Uploadcare)** includes the following fix.

==== Image previews sometimes showed the wrong image from a `srcset`.

> In previous version of the Image Optimizer premium plugin, an issue where image effects were applied based on the main `src` instead of the active `srcset` URL, causing discrepancies when the displayed image was selected from `srcset`. This led to a mismatch in the adjustment preview (rendered from the main `src`) versus the image shown in the editor (rendered from the `srcset` URL), resulting in different visual outcomes for effects such as blur or resize.

Reviewer: ```suggestion
In previous versions of the Image Optimizer premium plugin, an issue where image effects were applied based on the main `src` instead of the active `srcset` URL, caused discrepancies when the displayed image was selected from `srcset`. This led to a mismatch in the adjustment preview (rendered from the main `src`) versus the image shown in the editor (rendered from the `srcset` URL), resulting in different visual outcomes for effects such as blur or resize.
```

**Revision History** includes the following fix and improvement.

==== The same user could receive two different default avatars
// #TINY-12527

In previous versions of Revision History, the same user could receive two different default avatars. This inconsistency could make a single user appear as two distinct contributors, creating confusion when reviewing revision logs.

In {productname} {release-version}, the issue has been resolved so that each user is now always assigned the same default avatar, improving accuracy and consistency in the revision history display.

==== Default avatars are now generated with a consistent color based on the `user_id`
// #TINY-12527

Default avatars are now generated consistently using the `user_id`, ensuring that users are assigned a reliable and predictable color for their avatar. This improvement enhances the user experience by providing clear visual consistency when identifying contributors in revision histories.

For information on the **Revision History** plugin, see: xref:revisionhistory.adoc[Revision History].

=== Comments

The {productname} {release-version} release includes an accompanying release of the **Comments** premium plugin.

> **Comments** Premium plugin includes the following fixes and improvement and removal. 

Reviewer: ```suggestion
**Comments** Premium plugin includes the following fixes and improvement. 
```

.Example: how to use the `+getContent+` API to retrieve content with different formatting options.
[source,js]
----
// Disable indentation
tinymce.get(0).getContent({ indent: false });

// Serialize using named entities
tinymce.get(0).getContent({ indent: false, entity_encoding: 'named' });

// Serialize using numeric entities
tinymce.get(0).getContent({ indent: false, entity_encoding: 'numeric' });

// Serialize core entities as named, others as numeric
tinymce.get(0).getContent({ indent: false, entity_encoding: 'named+numeric' });

// Output raw characters
tinymce.get(0).getContent({ indent: false, entity_encoding: 'raw' });
----

or

.Example of configuring the editor to use named entity encoding and disable indentation.
[source,js]
----
tinymce.init({
  selector: "textarea",
  entity_encoding: 'named', // or 'numeric' or 'raw'
  indent: false, // or true
});
----

For more information on the `+getContent+` API, see `xref:apis/tinymce.editor.adoc#getContent[getContent()]`.

> === Clicking on a non selectable element when the selection is off screen no longer scrolls to the selection

Reviewer: ```suggestion
=== Clicking on a non-selectable element when the selection is off screen no longer scrolls to the selection
```

// #TINY-12245

> Previously, and issue was identified where clicking the editor's horizontal scrollbar a "non-selectable" UI element implicitly focused the editor, placed the caret at the beginning of the content, and then scrolled the document vertically to that caret position, causing the page to scroll away from the user's current view.

Reviewer: ```suggestion
Previously, an issue was identified where clicking the editor's horizontal scrollbar a "non-selectable" UI element implicitly focused the editor, placed the caret at the beginning of the content, and then scrolled the document vertically to that caret position, causing the page to scroll away from the user's current view.
```

In {release-version}, the behavior of `+Command (CMD)+` + `+Backspace+` has been updated to match other deletion methods, ensuring that inline formatting is preserved.

=== Navigating between elements with `+contenteditable="true"+` was not possible on Firefox using the keyboard.
// #TINY-12459

In {productname} {release-version}, a Firefox-specific limitation prevented caret movement between editable elements when navigating with the keyboard. The caret became trapped inside a +figcaption+, blocking vertical navigation using the Arrow Down key and preventing users from moving to the next element.

This issue disrupted accessibility and editing flow for Firefox users relying on keyboard interactions. To resolve this, a new caret navigation module was introduced and integrated into the Firefox-specific arrow key handling logic. The solution works around a long-standing Firefox bug by programmatically advancing the caret out of blocked states, ensuring smoother and more consistent keyboard navigation across editable elements.

=== Cursor movement did not operate correctly after a `figure` was selected
// #TINY-12458

> In previous versions of {productname}, using the TAB key to move focus into a `figcaption` element after selecting a `figure` caused the editor to enter a broken state where the caret was displayed inside the figcaption but typing did not produce any visible content. This issue prevented users from editing figcaptions using only the keyboard, forcing them to rely on a mouse or other pointing device to make changes.

Reviewer: ```suggestion
In previous versions of {productname}, using the `TAB` key to move focus into a `figcaption` element after selecting a `figure` caused the editor to enter a broken state where the caret was displayed inside the figcaption but typing did not produce any visible content. This issue prevented users from editing figcaptions using only the keyboard, forcing them to rely on a mouse or other pointing device to make changes.
```

// placeholder here.

=== The `content_css_cors` configuration option has been deprecated
// #TINY-12578

As of {productname} {release-version}, the `content_css_cors` configuration option has been marked as *deprecated*. It will be completely removed in the upcoming {productname} 9.0 release. As an alternative, we recommend using xref:tinymce-and-cors.adoc#crossorigin[crossorigin] instead.

[[known-issues]]
== Known issues

This section describes issues that users of {productname} {release-version} may encounter and possible workarounds for these issues.

There is one known issue in {productname} {release-version}.

=== Full Page HTML plugin incompatibility with Suggested Edits and Revision History

> The Full Page HTML plugin enables document properties and metadata to be set. However, these are not picked up by Suggested Edits and Revision History, and actions performed in these plugins will overwrite the values set by the Full Page plugin.

Reviewer: ```suggestion
The Full Page HTML plugin enables document properties and metadata to be set. However, these are not picked up by Suggested Edits and Revision History, and actions performed in these plugins will overwrite the values set by the Full Page HTML plugin.
```
